xbmc/GUIPassword.cpp

```diff
@@ -448,7 +516,27 @@ bool CGUIPassword::CheckMenuLock(int iWindowID)
   if (bCheckPW)
     return IsMasterLockUnlocked(true); //Now let's check the PW if we need!
   else
-    return true;
+      // ...don't forget to check PW of sources as well
+      if (strMediasourcePath != "")
+  {
```

Comment:
Somethings obviously gone wrong here. Probably have to add {} after the else to encompass this whole block correctly.

Reply:
doesn't work with {} either